### PR TITLE
CASH-661 Update style and functionality of KvPillToggle

### PR DESCRIPTION
**Pull request by foochris**

### CODE REVIEW

File: src/components/Kv/KvPillToggle.vue [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-			v-for="option in options" :key="option.key"
+	<div class="pill-toggle">
+		<span
+			v-for="(option, index) in options" :key="option.key"
 			@click="pillToggled(option.key)"
-			class="pill-option"
-			:class="{ active: selected === option.key }">
-			<h3>
+			:class="[
+				'pill-option',
```
Comment (ksekhri) [inline src/components/Kv/KvPillToggle.vue]: Let's use `first-child` and `last-child` for this instead of array indices

Comment (foochris) [reply src/components/Kv/KvPillToggle.vue]: @ksekhri PR updated with `first-child` and `last-child` and other minor SASSy changes.

File: src/components/Kv/KvPillToggle.vue [hunk 2]
```diff
@@ -26,45 +26,71 @@ export default {
...
+	width: 100%;
 
-.pill-option:first-child {
-	border-top-left-radius: $form-border-radius;
-	border-bottom-left-radius: $form-border-radius;
-}
+	span {
+		background: #F7F7F7;
+		border: rem-calc(1) solid #C4C1C6;
+		color: #808080;
```
Comment (mcstover) [inline src/components/Kv/KvPillToggle.vue]: What are the vars for these colors?

File: src/components/Kv/KvPillToggle.vue [hunk 3]
```diff
@@ -26,45 +26,71 @@ export default {
...
+			border-bottom-left-radius: 0;
+			border-bottom-right-radius: 0;
+			border-top-left-radius: 0;
+			border-top-right-radius: 0;
+		}
+	}
 
+	.active {
+		background: #8CCB8C;
+		border: rem-calc(1) solid #4FAF4E;
```
Comment (mcstover) [inline src/components/Kv/KvPillToggle.vue]: vars

File: src/components/Kv/KvPillToggle.vue (hunk 2 above)
Comment (foochris) [reply src/components/Kv/KvPillToggle.vue]: @mcstover Unfortunately, there are no vars for the L54-56 vars.

Comment (mcstover) [reply src/components/Kv/KvPillToggle.vue]: Please create in _settings.scss

Comment (foochris) [reply src/components/Kv/KvPillToggle.vue]: @mcstover Color vars added to `_settings.scss` in a new block.

File: src/components/Kv/KvPillToggle.vue (hunk 3 above)
Comment (foochris) [reply src/components/Kv/KvPillToggle.vue]: @mcstover Color vars added to `_settings.scss` in a new block.

File: src/assets/scss/_settings.scss [hunk 4]
```diff
@@ -1002,6 +1002,13 @@ $tab-content-border: $light-gray;
 $tab-content-color: $body-font-color;
 $tab-content-padding: 1rem;
 
+// 51.1 Tabs - Pill - vars added by Kiva
+// ------------------
+$tab-pill-border: rem-calc(1) solid #c4c1c6;
+$tab-pill-background: #f7f7f7;
+$tab-pill-color: #808080;
+$tab-pill-active-background: #8ccb8c;
+
```
Comment (ksekhri) [inline src/assets/scss/_settings.scss]: Nice

File: src/components/Kv/KvPillToggle.vue [hunk 5]
```diff
@@ -26,45 +26,71 @@ export default {
 	},
 	methods: {
 		pillToggled(key) {
-			this.$emit('pillToggled', key);
+			this.selected = key;
```
Comment (ksekhri) [inline src/components/Kv/KvPillToggle.vue]: Couple of notes here:
1. This is actually an anti-pattern in Vue. We only pass data down the component tree using props, not back up. This causes a console error warning us: `Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed property based on the prop's value. Prop being mutated: "selected"`. Basically whenever our parent re-renders any changes made by this line of code will be reset. @emuvente shared a useful link with more info here: http://michaelnthiessen.com/avoid-mutating-prop-directly/
2. As this is a controlled component, we'll update this value in the parent. We can make other changes to make this a mixed (controlled/uncontrolled) component too but for now we can remove this line and read the `pillToggled` event in the parent.

tl;dr: We can delete line 29

File: src/components/Kv/KvPillToggle.vue [hunk 6]
```diff
@@ -26,45 +26,71 @@ export default {
...
+			text-align: center;
+			width: 100%;
+		}
 
-.pill-option.active {
-	border: 1px solid $kiva-darkgreen;
-	cursor: default;
-}
+		&:first-child {
+			border-bottom-left-radius: $form-border-radius;
```
Comment (ksekhri) [inline src/components/Kv/KvPillToggle.vue]: Nice!

File: src/components/Kv/KvPillToggle.vue [hunk 7]
```diff
@@ -26,45 +26,71 @@ export default {
...
-}
+		&:last-child {
+			border-bottom-right-radius: $form-border-radius;
+			border-top-right-radius: $form-border-radius;
+		}
 
-.pill-option.active h3 {
-	background-color: $kiva-green;
-}
+		&:not(:first-child):not(:last-child) {
```
Comment (ksekhri) [inline src/components/Kv/KvPillToggle.vue]: Can we move these lines 82-25 up under line 60 (under selector `.pill-toggle span`)? It'll have the same effect up there.

**Comment by ksekhri** [comment]
@BoulderBrains I think this should work with your code fine, no interface changes

File: src/components/Kv/KvPillToggle.vue (hunk 7 above)
Comment (foochris) [reply src/components/Kv/KvPillToggle.vue]: @ksekhri PR updated per your suggestion.

File: src/components/Kv/KvPillToggle.vue (hunk 5 above)
Comment (foochris) [reply src/components/Kv/KvPillToggle.vue]: @ksekhri PR updated per your suggestion.

File: src/components/Kv/KvPillToggle.vue (hunk 2 above)
Comment (mcstover) [reply src/components/Kv/KvPillToggle.vue]: Thank you!

**Comment by BoulderBrains** [comment]
bitmoji

File: src/components/Kv/KvPillToggle.vue [hunk 8]
```diff
@@ -26,45 +26,61 @@ export default {
...
+			border-top-right-radius: $form-border-radius;
+		}
+	}
 
-.pill-option.active h3 {
-	background-color: $kiva-green;
+	.active {
+		background: $tab-pill-active-background;
+		border: rem-calc(1) solid $kiva-green;
+		color: $white !important;
```
Comment (ksekhri) [inline src/components/Kv/KvPillToggle.vue]: We should remove `!important` here

Comment (foochris) [reply src/components/Kv/KvPillToggle.vue]: @ksekhri PR updated with removal of `!important`
